bundles/org.openhab.core.auth.oauth2client/src/main/java/org/openhab/core/auth/oauth2client/internal/OAuthClientServiceImpl.java

```diff
@@ -150,7 +150,7 @@ public String getAuthorizationUrl(@Nullable String redirectURI, @Nullable String
             throw new OAuthException("Missing client ID");
         }
 
-        OAuthConnector connector = new OAuthConnector(httpClientFactory);
+        OAuthConnector connector = new OAuthConnector(httpClientFactory, persistedParams.deserializerClassName);
```

Comment:
A minor reminder for later refactoring: This looks like a potential candidate to be used as singleton.

Reply:
> 
> 
> A minor reminder for later refactoring: This looks like a potential candidate to be used as singleton.

Sorry, I'm afraid I did not get your point here